packages/amplify-provider-awscloudformation/src/graphql-resource-manager/amplify-graphql-resource-manager.ts

```diff
@@ -304,16 +317,67 @@ export class GraphQLResourceManager {
       const ddbStack = this.getStack(tableMeta.stackName, currentState);
       this.dropTemplateResources(ddbStack);
-
       // clear any other states created by GSI updates as dropping and recreating supersedes those changes
       this.clearTemplateState(tableMeta.stackName);
       this.templateState.add(tableMeta.stackName, JSONUtilities.stringify(ddbStack));
     });
+
+    /**
+     * When rebuild api, the root stack needs to change the reference to nested stack output values to temporary null placeholder value
+     * as there will be no output from nested stacks.
+     */
+    if (this.rebuildAllTables) {
+      const rootStack = this.getStack(ROOT_LEVEL, currentState);
+      const connectionStack = this.getStack(CONNECTION_STACK_NAME, currentState);
+      const searchableStack = this.getStack(SEARCHABLE_STACK_NAME, currentState)
+      const allRecreatedNestedStackNames = recreatedTables.map(tableMeta => tableMeta.stackName);
+      // Drop resources and outputs for connection stack if existed
+      if (connectionStack) {
+        allRecreatedNestedStackNames.push(CONNECTION_STACK_NAME);
+        this.dropTemplateResources(connectionStack);
+        this.templateState.add(CONNECTION_STACK_NAME, JSONUtilities.stringify(connectionStack));
+      }
+      // Drop resources and outputs for searchable stack if existed
+      if (searchableStack) {
+        allRecreatedNestedStackNames.push(SEARCHABLE_STACK_NAME);
+        this.dropTemplateResources(searchableStack);
+        this.templateState.add(SEARCHABLE_STACK_NAME, JSONUtilities.stringify(searchableStack));
+      }
+      // Update nested stack params in root stack
+      this.replaceRecreatedNestedStackParamsInRootStackTemplate(allRecreatedNestedStackNames, rootStack);
+      this.templateState.add(RESERVED_ROOT_STACK_TEMPLATE_STATE_KEY_NAME, JSONUtilities.stringify(rootStack));
+    }
   };
 
+  /**
+   * Set recreated nested stack parameters to 'TemporaryPlaceholderValue' in root stack template
+   * @param recreatedNestedStackNames names of recreated stacks
+   * @param rootStack root stack template
+   */
+  private replaceRecreatedNestedStackParamsInRootStackTemplate(recreatedNestedStackNames: string[], rootStack: Template) {
+    recreatedNestedStackNames.forEach(stackName => {
+      const stackParamsMap = rootStack.Resources[stackName].Properties.Parameters;
+      Object.keys(stackParamsMap).forEach(stackParamKey => {
+        const paramObj = stackParamsMap[stackParamKey];
+        const paramObjKeys = Object.keys(paramObj);
+        if ( paramObjKeys.length === 1 && paramObjKeys[0] === 'Fn::GetAtt') {
+          const paramObjValue  = paramObj[paramObjKeys[0]]
+          if ( Array.isArray(paramObjValue)
+            && paramObjValue.length === 2
+            && recreatedNestedStackNames.includes(paramObjValue[0])
+            && paramObjValue[1].startsWith('Outputs.') ) 
+          {
+            stackParamsMap[stackParamKey] = 'TemporaryPlaceholderValue';
```

Comment:
I don't fully understand the fix here. Can you post a screenshot of before and after state of a nested child stack parameters? How are we ensuring that these defaults will not show up in the final templates deployed via `push`?
Instead of adding a default can we just remove these parameters?

Reply:
The parameters cannot be removed but faking a placeholder value. These parameter values are defined in api root stack (`cloudformation-template.json` => `Resources.[nestedStack].Properties.Parameters`). I update a example in the PR description.

Reply:
It looks like we are looking for a very specific CFN template string. Could this break if there is a change in the way we pass this parameter value in the future?

Reply:
The phrase it looks for is the `Fn:GetAtt:[Outputs.*, ]` which is an intrinsic function and reserved word in the Cloudformation. The words should be unlikely to change unless the breaking change occurs in CFN. 
Currently it is constructed by the CDK sync operation. If it changes its form, this will be a breaking change for CDK users and the CDK should convey this message to us.